crates/optimism/node/src/txpool.rs

```diff
@@ -269,7 +267,7 @@ impl<Client, Tx> OpTransactionValidator<Client, Tx> {
 
 impl<Client, Tx> OpTransactionValidator<Client, Tx>
 where
-    Client: StateProviderFactory + BlockReaderIdExt,
+    Client: StateProviderFactory + BlockReaderIdExt<Block = OpBlock>,
```

Comment:
Ah I see there's an associated type on the `BlockReaderIdExt` that's good to know

Reply:
yerp, this is the lowest level change in the reth sdk / remove optimism feature project! basically since op has deposit tx type, we needed to make the whole codebase generic over the blockchain data primitives: transaction, receipt, tx type, etc.

this is still wip afaik, so some places in the codebase have hardcoded constraints on the ATs, some don't. this is why you sometimes get the error:
```
expected: alloy_consensus::Header
got: <SomeTrait>::Header
```

Reply:
@refcell for visibility